expsync/database.go

```diff
@@ -0,0 +1,58 @@
+package expsync
+
+import (
+	"app/base/models"
+	"encoding/json"
+
+	"gorm.io/gorm/clause"
+
+	"gorm.io/gorm"
+)
+
+var (
+	DB *gorm.DB
+)
+
+func updateExploitsMetadata(tx *gorm.DB, exploits map[CVE][]ExploitMetadata) (int64, error) {
+	if len(exploits) == 0 {
+		return 0, nil
+	}
+
+	cves := make([]models.Cve, 0, len(exploits))
+
+	for cve, metadata := range exploits {
+		exploitBytes, err := json.Marshal(metadata)
+		if err != nil {
+			return 0, err
+		}
+
+		cves = append(cves, models.Cve{
+			Name: string(cve), ExploitData: exploitBytes,
+			// Satisfy GORM required columns check for Description and Severity without performing update.
+			Description: "desc", Severity: models.NotSet,
+		})
+	}
+
+	res := tx.Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "name"}},
+		DoUpdates: clause.AssignmentColumns([]string{"exploit_data"}),
+	}).Create(cves)
```

Comment:
I'd change this job to only do the update (if the CVE exists in DB). If the CVE isn't in the database then not create it. This is how it's implemented in https://github.com/RedHatInsights/vulnerability-engine/blob/master/exploit_sync/exploit_sync.py